### PR TITLE
Implement gui

### PR DESCRIPTION
**Pull request by david-ribeiro-sumup**
Implemented a simple GUI to demo the terminal API. Refactored code to be usable both as a script and on the UI. Updated README.md with instructions.

### CODE REVIEW

**Comment by chad-lung-fivestars** [comment]
I have some feedback for this. We can review that today.

File: terminal-api/utils.ts [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+
+export async function httpRequest(
+  endpoint: string,
+  method: string,
+  body: string | null,
+  config: ConfigurationSchema,
+  timeout = 120000
+): Promise<Response> {
+  let response: Response;
+
```
Comment (chad-lung-fivestars) [inline terminal-api/utils.ts:66]: The "delay" should be put in here and removed from the other spots. We want the delay to slow all calls using httpRequest()

**Comment by chad-lung-fivestars** [comment]
The latest update looks really good. The one thing I noticed is that you can't cancel. You should be able to cancel anytime from the customer checking in up until the payment is started.

**Comment by chad-lung-fivestars** [comment]
FYI - I was only able to apply a reward once. After that I couldn't select cash/credit/other in my next transaction.

**Comment by david-ribeiro-sumup** [comment]
> The latest update looks really good. The one thing I noticed is that you can't cancel. You should be able to cancel anytime from the customer checking in up until the payment is started.

The API to cancel needs the pos-checkout-id, so I guess I can't cancel until I have one. Am I missing something?

**Comment by david-ribeiro-sumup** [comment]
> FYI - I was only able to apply a reward once. After that I couldn't select cash/credit/other in my next transaction.

I think I fixed it in the commit I just made

**Review by chad-lung-fivestars — APPROVED** [review]
👍  This looks great now and cancel is working as expected. Please merge when you get a chance.

**Pull request merged**